src/Path/Include.hs

```diff
@@ -409,6 +408,61 @@ infixr 7 -<.>
   -> m (Path b File)   -- ^ New file name with the desired extension
 (-<.>) = flip setFileExtension
 
+-- | Tries to remove specific extension from given file path. Throws if the
+-- resulting filename does not parse.
+--
+-- >>> stripFileExtension "txt" $(mkRelFile "foo")
+-- Nothing
+-- >>> stripFileExtension "baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo.bar"
+-- >>> stripFileExtension ".baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo.bar"
+-- >>> stripFileExtension "bar.baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo"
+-- >>> stripFileExtension ".bar.baz" $(mkRelFile "foo.bar.baz")
+-- Just "foo"
+-- >>> stripFileExtension "foo.bar.baz" $(mkRelFile ".foo.bar.baz")
+-- *** Exception: InvalidRelFile ""
+--
+-- @since 0.6.2
+stripFileExtension
+  :: MonadThrow m
+  => String                  -- ^ Extension to strip
+  -> Path b File             -- ^ Old file name
+  -> m (Maybe (Path b File)) -- ^ New file name without extension
```

Comment:
This should be just `m (Path b File)`. The `Maybe` is unnecessary because we have `MonadThrow m`.

Reply:
The `filepath` package returns Nothing if extension didn't match. It seems a bit too much to throw exception in such a common case.

Reply:
The `Maybe` instance of `MonadThrow` takes care of that:

```
instance MonadThrow Maybe where
  throwM _ = Nothing
```

So we have:

```
Prelude Path Control.Monad.Catch> f = throwM (InvalidAbsDir "x") :: MonadThrow m => m Int
Prelude Path Control.Monad.Catch> f :: Maybe Int
Nothing
```

Take a look at `addFileExtension` it also throws an exception. We need exactly the same signature here as well.

Reply:
Sorry, I wasn't clear enough. If Maybe will be removed then an exception in `MonadThrow m => m` will be used so signal either that resulting path is invalid or that suffix didn't match. If both these use cases are merged into single `MonadThrow m` then user may choose `Maybe` for `m` and get Nothing when resulting path is invalid and doesn't parse or using `m` as `IO` or something similar he'll get an exception when the suffix was not present. I believe that if resulting path is invalid then a loud error/exception should be produced.

Perhaps function's semantics should be changed to always return either the original path or a path without a given extension.

Reply:
We do not have two distinct cases here. Either we are able to strip the extension, resulting in a valid filepath and extension (Just case), or not (Nothing case). I cannot think of a case where we have to distinguish the error case in two different categories. Am I missing something?

Reply:
Looking at the test cases, this is the case where you are throwing an exception:

```
stripFileExtension ".foo" $(mkRelFile ".foo")
```

There are two ways of looking at this. First is to treat ".foo" as an extension to an invalid filename i.e. an empty filename. The second is to treat ".foo" as a filename without an extension rather than an extension, the "." is part of the filename in this case rather than being an extension separator. You have chosen the former way. I think we can choose the latter and keep the API simple. Two layers of error handling makes the API unnecessarily complex.